socorro/processor/mozilla_transform_rules.py

```diff
@@ -280,12 +281,21 @@ class JavaProcessRule(Rule):
     def version(self):
         return '1.0'
 
+    def _predicate(self, raw_crash, raw_dumps, processed_crash, proc_meta):
+        return bool(raw_crash.get('JavaStackTrace', None))
+
     def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        # This can contain PII in the exception message
+        processed_crash['java_stack_trace_full'] = raw_crash['JavaStackTrace']
 
-        processed_crash.java_stack_trace = raw_crash.setdefault(
-            'JavaStackTrace',
-            None
-        )
+        try:
+            java_exception = javautil.parse_java_stack_trace(raw_crash['JavaStackTrace'])
+            java_stack_trace = java_exception.to_public_string()
+        except javautil.MalformedJavaStackTrace as mjst:
+            processor_meta.processor_notes.append('JavaProcessRule: %r' % mjst)
```

Comment:
I was careful to make sure `MalformedJavaStackTrace` never has an exception message that contains anything interesting in it. It's either a string from the code or it's whatever the ValueError message is if there's no `:` in the first line.

Now that I'm rethinking this, that code is so far away we should assume someone will change it breaking this assumption. I'll fix this.